torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -258,6 +258,23 @@ void OwnerRRef::setError(std::exception_ptr eptr) {
   future_->setErrorIfNeeded(std::move(eptr));
 }
 
+// #ifdef USE_CUDA_NOT_ROCM
+void OwnerRRef::recordAllDevices(const std::set<c10::DeviceIndex>& deviceIndices) {
+  cudaEvents_.clear();
+  for (auto deviceIndex : deviceIndices) {
+    at::cuda::CUDAEvent cudaEvent;
+    cudaEvent.record(at::cuda::getCurrentCUDAStream(deviceIndex));
```

Comment:
yep, it's OK to record on current streams, as the agent has set up current streams using the ctx.